crates/uv/src/requirements.rs

```diff
@@ -281,15 +281,15 @@ impl RequirementsSpecification {
         constraints: &[RequirementsSource],
         overrides: &[RequirementsSource],
         extras: &ExtrasSpecification<'_>,
-        connectivity: Connectivity,
+        client_builder: BaseClientBuilder,
     ) -> Result<Self> {
         let mut spec = Self::default();
 
         // Read all requirements, and keep track of all requirements _and_ constraints.
         // A `requirements.txt` can contain a `-c constraints.txt` directive within it, so reading
         // a requirements file can also add constraints.
         for source in requirements {
-            let source = Self::from_source(source, extras, connectivity).await?;
+            let source = Self::from_source(source, extras, client_builder.clone()).await?;
```

Comment:
These clones are alarming to me. We should pass a reference to a lazily built client instead?

Reply:
Addressed here in https://github.com/astral-sh/uv/pull/2434/commits/2bcf617d6a4fe43538e2ceb4209178542495b726 instead